src/si/time.rs

```diff
@@ -76,11 +77,10 @@ mod tests {
 
         #[test]
         fn from() {
-            // let t = Time::new::<second>(21.5);
             let t = Time::new::<second>(-21.5);
             let d: Duration = t.into();
-            assert_eq!(21, d.as_secs());
-            // assert_eq!(5e8 as u32, d.subsec_nanos());
+            assert_eq!(d.as_secs(), 21);
+            assert!(499_999_999 <= d.subsec_nanos() && d.subsec_nanos() <= 500_000_001);
```

Comment:
Is there a reason that we can't require this to be exactly 500,000,000?

Reply:
Precision loss when I subtract t from zero is my current best guess.  The way I implemented it originally, by casting the time to f64 and then converting that to a Duration did allow this to be 500,000,000 exactly.  Perhaps it's only the f32 test that loses precision?